src/AddItems.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import { db } from './lib/firebase';
+import { capitalize } from './lib/utility';
+
+function AddItems(props) {
+  const [groceryItem, setGroceryItem] = useState('');
+  const [userToken, setUserToken] = useState('randomTokenOne');
+  const [itemFreq, setItemFreq] = useState(7);
+  const [formValidationVisual, setFormValidationVisual] = useState(false);
+
+  const updateGroceryItem = (event) => {
+    setGroceryItem(capitalize(event.target.value));
+    setFormValidationVisual(true);
+  };
+
+  const submitGroceryItem = (event) => {
+    event.preventDefault();
+    const formData = {
+      itemName: groceryItem,
+      frequency: itemFreq,
+      lastPurchaseDate: null,
+    };
+    db.collection(userToken).add({ formData });
+    setGroceryItem('');
+    setItemFreq(7);
+    setFormValidationVisual(false);
+  };
+
+  const radioBtnHandler = (event) => {
+    setItemFreq(+event.target.value);
+    setFormValidationVisual(true);
+  };
+
+  return (
+    <div>
+      <form id="test-form" onSubmit={submitGroceryItem}>
+        <label htmlFor="item">Item name:</label>
+        <input
+          type="text"
+          id="item"
+          value={groceryItem}
+          required={formValidationVisual}
```

Comment:
I _think_ you can make this just `required` without having some kind of state value associated with it.

Reply:
Hey @laurenmbeatty! We originally did have the required without any state association but what I noticed was that once the first item is submitted, the text field gets that outline indicating the user has forgotten to fill a `required` field in the form.

[https://www.loom.com/share/13122feca4044035b037d955523a5400](url)

Reply:
I'm also curious to know why the `required` attribute is controlled by state.

I was able to submit a couple of items with blank text and I think it happened because `formValidationVisual` is initially set to `false`, meaning that the text field is not marked as required until I start typing in it: https://www.loom.com/share/a8b989c03da14e91b87f1d9dbc50aa92

So I'm just going to request a change to prevent the submission of blank items.

Reply:
That's really interesting, @kkatoch89, that the red outline appears. Do you think if you set formValidationVisual, what @dariatsvetkova was saying, that it would prevent that red outline as well?

Reply:
@laurenmbeatty @dariatsvetkova @ablizben We went ahead and reverted to just using `required` rather than tying it to `formValidationVisual` - we can decide as a team if the red outline is something we want to handle or not! 😊

Reply:
Hey @dariatsvetkova and @ablizben, great points! So my answers to those would be:
1. If you just set a simple `require` without any associated state like `formValidationVisual`, it would definitely prevent the form from submitting an empty input; though it will not prevent the user from submitting for example an input that has just whitespace in it. So we would have to still deal with a similar edge case as submitting a totally empty input.
2. To Abby's point, if I understand it correctly, the `formValidationVisual` does help prevent the red outline from appearing automatically on submit. I'd have to test it out but I think if I set `formValidationVisual` as true just before we execute our handle submit function, it could prevent an empty input field from submitting and display the red outline then.

Reply:
We've decided on going with the simple `require` for now and we can come up with a solution for the edge cases as a team going forward 😎😎😎😎😎😎😎